packages/jest-reporters/src/get_watermarks.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {Config} from '@jest/types';
+import istanbulReport = require('istanbul-lib-report');
+
+export default function getWatermarks(
+  config: Config.GlobalConfig,
+): istanbulReport.Watermarks {
+  const defaultWatermarks = istanbulReport.getDefaultWatermarks();
+
+  const {coverageThreshold} = config;
+
+  if (!coverageThreshold || !coverageThreshold.global) {
+    return defaultWatermarks;
+  }
+
+  const keys: Array<keyof Config.CoverageThresholdValue> = [
+    'branches',
+    'functions',
+    'lines',
+    'statements',
+  ];
+  return keys.reduce((watermarks, key) => {
+    const value = coverageThreshold.global[key];
```

Comment:
is it an issue that this only looks at `global` and not other defined thresholds?

Reply:
As far as I understand it istanbul is only used to generate coverage maps and coverage reports, it doesn't actually check anything. The checks are done by jest [here]( https://github.com/facebook/jest/blob/master/packages/jest-reporters/src/coverage_reporter.ts#L238). 

I tried to see if istanbul could take different watermarks for different paths but I found nothing short of running istanbul once for every coverage threshold group.

Reply:
Aight, cool. Thanks for checking 🙂